swimlane/core/resources/task.py

```diff
@@ -0,0 +1,18 @@
+from swimlane.core.resources.base import APIResource
+
+
+class Task(APIResource):
+    _type = 'Core.Models.Integrations.Task, Core'
+
+    def __init__(self, swimlane, raw):
+        super(Task, self).__init__(swimlane, raw)
+        self.app_id = raw.get('applicationId')
+        self.id = raw.get('id')
+        self.name = raw.get('name')
+        self.script = raw.get('action').get('script')
```

Comment:
Is it used anywhere? Not sure enrich is actually required in this context.

Reply:
It is not currently used anywhere. I initially thought it may be useful to have if someone wanted to update or make modifications to action scripts programmatically. I have some other ideas for the Task resource and this might have bled over into that. I can remove it if you want me to.

Reply:
It would be great. We try not to slow down performance or put more code before it becomes necessary. Half of the time it never gets implemented the way it was thought out. We also have idea to create public API which will be more like python driver. So python driver will be more 1:1 to it.

Reply:
After reviewing it again, It is used in the hash function. One use case would be getting Tasks between two Swimlane instances and comparing them; another use case could be trying to import a task from JSON and need to know if the task is similar from the script standpoint. However, it doesn't account for other changes that may exist, inputs/outputs/triggers, etc. Curious to hear you're thoughts on this use, would there be a better way to compare two task objects?

Reply:
> `self.script = raw.get('action').get('script')`

My org has scripts which connect to MongoDB and dump Tasks into a GitLab repo for change tracking over time (and prefix scripts with comments relating to triggers/etc). We eventually plan on using PRs and pipelines to review and push from a dev instance to a prod instance.

This functionality would be super helpful as it takes the manual correlation out of Mongo-level `id` correlations.

Cheers.

Reply:
Ids should be unique

Reply:
@VC

> > `self.script = raw.get('action').get('script')`
> 
> My org has scripts which connect to MongoDB and dump Tasks into a GitLab repo for change tracking over time (and prefix scripts with comments relating to triggers/etc). We eventually plan on using PRs and pipelines to review and push from a dev instance to a prod instance.
> 
> This functionality would be super helpful as it takes the manual correlation out of Mongo-level `id` correlations.
> 
> Cheers.

So it is need perfect.

Reply:
> > `self.script = raw.get('action').get('script')`
> 
> 
> 
> My org has scripts which connect to MongoDB and dump Tasks into a GitLab repo for change tracking over time (and prefix scripts with comments relating to triggers/etc). We eventually plan on using PRs and pipelines to review and push from a dev instance to a prod instance.
> 
> 
> 
> This functionality would be super helpful as it takes the manual correlation out of Mongo-level `id` correlations.
> 
> 
> 
> Cheers.

Thanks for the feedback. That was also a use case I had in mind. We're looking to implement a CI/CD pipeline as well and potentially export the scripts in a task to a repo to track changes over time.

The other benefit of these changes, and what drove me to make them, was to be able to write functional tests for tasks to use for automated testing, i.e. Create a record, populate fields of the record that a task is and input of a task then execute it against the record and compare the expected outputs to the record. If task execution fails it throws an exception, if successful the record is updated with current state.